src/main/kotlin/no/nav/familie/ks/sak/kjerne/overgangsordning/OvergangsordningAndelService.kt

```diff
@@ -42,11 +49,38 @@ class OvergangsordningAndelService(
         val overgangsordningAndel = finnOvergangsordningAndel(overgangsordningAndelId)
         val personopplysningGrunnlag = personopplysningGrunnlagService.hentAktivPersonopplysningGrunnlagThrows(behandling.id)
         val person = personopplysningGrunnlag.personer.single { it.aktør.aktivFødselsnummer() == overgangsordningAndelRequestDto.personIdent }
-        val vilkårsvurdering = vilkårsvurderingService.hentAktivVilkårsvurderingForBehandling(behandling.id)
+        val vilkårsvurdering by lazy { vilkårsvurderingService.hentAktivVilkårsvurderingForBehandling(behandling.id) }
+        val andreAndelerPåBehandling by lazy { hentOvergangsordningAndeler(behandling.id).filter { it.id != overgangsordningAndelId } }
+
+        val utfyltOvergangsordningAndel =
+            overgangsordningAndel
+                .fraOvergangsordningAndelDto(overgangsordningAndelRequestDto, person)
+                .tilUtfyltOvergangsordningAndel()
+
+        val barnehageplassVilkår =
+            vilkårsvurdering
+                .hentPersonResultaterTilAktør(person.aktør.aktørId)
+                .filter { it.vilkårType == Vilkår.BARNEHAGEPLASS }
+
+        validerFomDato(
+            andel = utfyltOvergangsordningAndel,
+            gyldigFom = beregnGyldigFom(person),
+        )
 
-        overgangsordningAndel.fraOvergangsordningAndelDto(overgangsordningAndelRequestDto, person)
+        validerTomDato(
+            andel = utfyltOvergangsordningAndel,
+            gyldigTom = beregnGyldigTom(person),
+        )
+
+        validerIngenOverlappMedEksisterendeOvergangsordningAndeler(
+            nyAndel = utfyltOvergangsordningAndel,
+            eksisterendeAndeler = andreAndelerPåBehandling,
+        )
 
-        // TODO: Validering
+        validerAtBarnehagevilkårErOppfyltIOvergangsordningAndelPeriode(
+            overgangsordningAndel = utfyltOvergangsordningAndel,
+            barnehageplassVilkår = barnehageplassVilkår,
```

Comment:
I valideringsmetodene brukes forskjellig parameternavn på overgangsordningAndel (andel, nyAndel overgangsordningAndel). Kanskje greit å bare bruke overgangsordningAndel for alt?